perllib/FixMyStreet/Cobrand/Westminster.pm

```diff
@@ -70,6 +75,63 @@ sub open311_config {
     my $id = $row->user->get_extra_metadata('westminster_account_id');
     # Westminster require 0 as the account ID if there's no MyWestminster ID.
     $h->{account_id} = $id || '0';
+
+    my $extra = $row->get_extra_fields;
+
+    # Reports made via the app probably won't have a USRN because we don't
+    # display the road layer. Instead we'll look up the closest asset from the
+    # asset service at the point we're sending the report over Open311.
+    if (!$row->get_extra_field_value('USRN')) {
+        if (my $ref = $self->lookup_site_code($row)) {
+            push @$extra, { name => 'USRN', value => $ref };
+        }
+    }
+
+    $row->set_extra_fields(@$extra);
+}
+
+sub lookup_site_code_config {
+    # uncoverable subroutine
+    # uncoverable statement
+    {
+        buffer => 1000, # metres
+        proxy_url => "https://tilma.staging.mysociety.org/resource-proxy/proxy.php",
+        url => "https://westminster.assets/40/query",
+        property => "USRN",
+        accept_feature => sub { 1 }
+    }
+}
+
+sub _fetch_features {
```

Comment:
A lot of this is the same as what it's overwriting - a factored out `lookup_site_code_url` function would be better.

Reply:
Good idea, have added that in 334f4370d95649041120bc0e5e5858ba6edfbf1c.